src/groupByKey.ts

```diff
@@ -2,6 +2,29 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(object: T[], key: keyof T): GroupsMap<T> {
+  const finalObject: GroupsMap<T> = {};
+
+  object.forEach((el) => {
+    const keyValue = el[key];
+
+    if (keyValue !== undefined && keyValue !== null) {
+      const keyString = String(keyValue);
+
+      if (keyString !== '') {
```

Comment:
| think you need to check here your string with trim() 
```suggestion
      if (keyString.trim()) {
```